src/map/script.c

```diff
@@ -7496,6 +7496,86 @@ BUILDIN(getelementofarray)
 	return true;
 }
 
+BUILDIN(array_find)
+{
+	struct script_data *data = script_getdata(st, 2);
+	struct reg_db *ref = NULL;
+	const char *name;
+	int i = 0, id, start_index, array_size;
+
+	if (!data_isreference(data)) {
+		ShowError("buildin_array_find: not a variable\n");
+		script->reportdata(data);
+		script_pushnil(st);
+		st->state = END;
+		return false;
+	}
+
+	start_index = reference_getindex(data);
+	if (start_index < 0 || start_index > SCRIPT_MAX_ARRAYSIZE) {
+		ShowError("buildin_array_find: starting index out of range (%d)\n", start_index);
+		script->reportdata(data);
+		script_pushnil(st);
+		st->state = END;
+		return false;
+	}
+
+	id = reference_getid(data);
+	name = reference_getname(data);
+	ref = reference_getref(data);
+
+	array_size = script->array_highest_key(st, st->rid ? script->rid2sd(st) : NULL, name, ref);
+	if (array_size >= SCRIPT_MAX_ARRAYSIZE) { // edge case
+		ShowError("buildin_array_find: array size is too large.\n");
+		script->reportdata(data);
+		script_pushnil(st);
+		st->state = END;
+		return false;
+	}
+
+	if (is_string_variable(name)) {
+		const char *temp, *value;
+
+		if (script_isinttype(st, 3)) {
+			ShowError("buildin_array_find: both sides must be same integer or string type.\n");
```

Comment:
not necessarily, here you could convert your integer to a string with sprintf

Reply:
no,
if we treat this command as a shorten for-loop, it will look like this
```
prontera,155,185,5	script	kdjfhskfhd	1_F_MARIA,{
	setarray .@a,2,4,6,8,10;
	for ( .@i = 0; .@i < getarraysize(.@a); ++.@i )
		if ( .@a == "2" )
			break;
	end;
}
```
which cause this error
```
[Error]: script:op_2: invalid data for operator C_EQ
[Debug]: Data: number value=2
[Debug]: Data: string value="2"
[Debug]: Source (NPC): kdjfhskfhd at prontera (155,185)
```
if we silently convert integer into string means promoting bad scripting habits